src/server/scripts/EasternKingdoms/zone_westfall.cpp

```diff
@@ -289,6 +355,285 @@ class spell_westfall_despawn_jangolode_actor : public SpellScript
+    {
+        uint8 clueGainChance = 0;
+        switch (gossipListId)
+        {
+            case Gossip::GossipOption::GossipOption0:
+            {
+                clueGainChance = 25;
+                break;
+            }
+            case Gossip::GossipOption::GossipOption1:
+            {
+                clueGainChance = 75;
+                _bribeFailed = true;
+                break;
+            }
+            default:
+                return true;
+        }
+
+        me->RemoveNpcFlag(UNIT_NPC_FLAG_GOSSIP);
+        CloseGossipMenuFor(player);
+        if (roll_chance_i(clueGainChance))
+            GiveClue(player);
+        else
+        {
+            me->RemoveUnitFlag(UNIT_FLAG_IMMUNE_TO_PC);
+            AttackStart(player);
+        }
+
+        return false;
+    }
+
+    void GiveClue(Player* player)
+    {
+        if (!player)
+            return;
+
+        _events.CancelEventGroup(Events::MurderWasTheCaseThatTheyGaveMe::GroupOOC);
+        me->SetFacingToObject(player);
+
+        uint16 slot = player->FindQuestSlot(Quests::MurderWasTheCaseThatTheyGaveMe);
+
+        if (player->GetQuestSlotCounter(slot, 0) == 0)
+        {
+            player->CastSpell(player, Spells::HoboInformation1);
+            Talk(Text::HoboText::HoboSayClueOne, player);
+            me->DespawnOrUnsummon(12s);
+            return;
+        }
+
+        if (player->GetQuestSlotCounter(slot, 1) == 0)
+        {
+            player->CastSpell(player, Spells::HoboInformation2);
+            Talk(Text::HoboText::HoboSayClueTwo, player);
+            me->DespawnOrUnsummon(12s);
+            return;
+        }
+
+        if (player->GetQuestSlotCounter(slot, 2) == 0)
+        {
+            player->CastSpell(player, Spells::HoboInformation3);
+            Talk(Text::HoboText::HoboSayClueThree, player);
+            me->DespawnOrUnsummon(12s);
+            return;
+        }
+
+        if (player->GetQuestSlotCounter(slot, 3) == 0)
+        {
+            player->CastSpell(player, Spells::HoboInformation4);
+            Talk(Text::HoboText::HoboSayClueFour, player);
+            me->DespawnOrUnsummon(12s);
+            return;
+        }
+    }
+
+    void JustEngagedWith(Unit* who) override
+    {
+        me->SetAIAnimKitId(0);
+        _events.CancelEventGroup(Events::MurderWasTheCaseThatTheyGaveMe::GroupOOC);
+
+        if (!who->IsPlayer())
+            return;
+
+        Talk(_bribeFailed ? Text::HoboText::HoboAggroBribe : Text::HoboText::HoboAggroConvince, who);
+    }
+
+    void DamageTaken(Unit* /*killer*/, uint32& damage, DamageEffectType /*damageType*/, SpellInfo const* /*spellInfo = nullptr*/) override
+    {
+        if (!_flee && me->HealthBelowPctDamaged(20, damage))
+        {
+            _flee = true;
+            me->DoFleeToGetAssistance();
+            Talk(Text::HoboText::HoboFlee);
+        }
+    }
+
+    void JustDied(Unit* who) override
+    {
+        if (who->IsCreature() && who->ToCreature()->IsAIEnabled())
```

Comment:
iirc there's no need to check IsAIEnabled anymore in scripts when accessing AI since there's no way it's ever null, there's a tick in between where AI is filled with a "nullAI" type